kernel/src/hil/sensors.rs

```diff
@@ -14,7 +14,7 @@ pub trait TemperatureClient {
     ///
     /// - `value`: the most recently read temperature in hundredths of degrees
     /// centigrate.
-    fn callback(&self, value: usize);
+    fn callback(&self, value: i32);
```

Comment:
I think the pertinent question is: what does that Option represent? Looking at Lsm303agrI2C, the "odd" path is triggered when the i2c status is an `Err`. Perhaps instead of Option, we should forward `Result<Temp, SensorError>`? Clients don't have to be in the userspace, so there's no need to drop error information.

Reply:
I think this is a possible solution and you are right, I would suggest `Result<usize, ErrorCode>` to avoid adding another error type.

Reply:
Agreed, if we are changing it I think a `Result<i32, ErrorCode>` is worth adding

Reply:
I think we should do the simple thing for now. Negative temperatures are broken, so let's fix them. Then we can go back and revise temperature (and all the other sensors) to use a Result + ErrorCode scheme in a different PR.

Reply:
@brghena I disagree -- I think it would be better to bring this HIL closer to a standard form. The proposal is already adding a specific "unavailable" value, we should format it correctly.

Reply:
My thinking had been that this PR ended up sitting around because attempting to fix negative temperatures got derailed into adding results to temperature sensors. It definitely would be better to have the result interface, for sure, but it wasn't clear to me that it had to happen here.

Reply:
I didn't follow up on this because I had an impression (perhaps wrong) that there was no consensus on which interface was preferred (and I didn't really use the temp sensor for anything but testing).

Making this a Result is no problem at all, but my next two weeks are busy.

Reply:
Sounds good then. Thanks for the clarification!

Reply:
@dcz-self Ping!

Reply:
I didn't forget! Just overloaded with chores :(